en/guide/mavlink_2.md

```diff
@@ -37,3 +81,38 @@ if (mavlink_parse_char(MAVLINK_COMM_0, buf[i], &msg, &status)) {
 
 
 
+## Message Extensions {#message_extensions}
+
+The XML for messages can contain extensions that are optional in the protocol. 
```

Comment:
@LorenzMeier I thought the purpose of extensions is to identify the new additions in MAVLink 2.0 so that they aren't generated into code for MAVLink 1. "The XML for messages can contain extensions that are optional in the protocol. " does not really capture that - can you explain?

The extensions thing seems a bit like a wasted opportunity to specify versioning.

Reply:
Sure, discussion here: https://github.com/mavlink/mavlink/issues/919

With respect to the the decoding this is covered in the "The rules for extended messages" that follows the example. I will those up to make it more clear.